libbeat/autodiscover/providers/kubernetes/orchestrator.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kubernetes
+
+import (
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+func genOrchestratorFields(kubemeta common.MapStr, kind string) common.MapStr {
+	fields := common.MapStr{
+		"type": "kubernetes",
+	}
+	resource := common.MapStr{
+		"type": kind,
+	}
+	namespace, err := kubemeta.GetValue("namespace")
+	if err == nil {
+		fields.Put("namespace", namespace)
+	}
+
+	nodeMap, err := kubemeta.GetValue(kind)
+	if err == nil {
+		node, _ := nodeMap.(common.MapStr)
```

Comment:
Does this need to be cast/nil checked? If you want to do an unconditional cast, I'd remove the `_` -- but assuming the above `GetValue` call returns a `nil` you probably want to be more explicit about the cast and `nil` checking here.

Reply:
> Some comments/questions.
> 
>     * I'm thinking we should probably add orchestrator data to the processors as well:
>       
>       * https://github.com/elastic/beats/tree/master/libbeat/processors/add_kubernetes_metadata
>       * https://github.com/elastic/beats/tree/master/x-pack/libbeat/processors/add_nomad_metadata

Added

> 
>     * For the docker autodiscover/processor do you know if there's a way via Docker's API to detect if you're running in Swarm? If so maybe we should consider adding there as well? If it's too much of a heavy lift maybe we can add that in later as a separate enhancement though.

It looks like you might be able to "inspect swarm", and know if you are in a swarm or not.  But I'm thinking a separate enhancement might be best.

Reply:
> Does this need to be cast/nil checked? If you want to do an unconditional cast, I'd remove the `_` -- but assuming the above `GetValue` call returns a `nil` you probably want to be more explicit about the cast and `nil` checking here.

changed this so we don't get intermediate maps, cleaner all around, don't know what I was thinking.